File: backend/MealPrepper.Core/Entities/MealEntry.cs
namespace MealPrepper.Core.Entities;

public class MealEntry
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid RecipeId { get; set; }
    public DateTime Date { get; set; }
    public MealSlot MealSlot { get; set; }
    public decimal PortionMultiplier { get; set; } = 1;
    public User User { get; set; } = null!;
    public Recipe Recipe { get; set; } = null!;
}

File: frontend/src/types/mealEntry.ts
export type MealSlot = 'Breakfast' | 'Lunch' | 'Dinner' | 'Snack'
export const MEAL_SLOTS: MealSlot[] = ['Breakfast', 'Lunch', 'Dinner', 'Snack']

export interface MealEntry {
  id: string
  userId: string
  recipeId: string
  recipeName: string
  date: string        // "YYYY-MM-DDTHH:mm:ss" from backend — use .split('T')[0]
  mealSlot: MealSlot
  portionMultiplier: number
}

export interface CreateMealEntryRequest {
  recipeId: string
  date: string        // "YYYY-MM-DD"
  mealSlot: MealSlot  // string enum: "Breakfast" etc.
  portionMultiplier: number
}
